src/libAtomVM/synclist.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of AtomVM.
+ *
+ * Copyright 2022 Paul Guyot <pguyot@kallisys.net>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+ */
+
+#include "synclist.h"
+
+#include "sys.h"
+
+#ifndef AVM_NO_SMP
+
+void synclist_init(struct SyncList *synclist)
+{
+    synclist->lock = avmsmp_rwlock_create();
+    list_init(&synclist->head);
+}
+
+void synclist_destroy(struct SyncList *synclist)
```

Comment:
I feel like that most of this functions might be inlined, so we can get better optimization.
It's ok to keep this right now as is, as first iteration, but let's keep this in our mind.

Reply:
There is a circular dependency issue that prevented from inlining these in synclist.h. sys.h depends on headers that depend on synclist.h

Reply:
Actually, now that SMP support system functions are in their own header, `smp.h`, this can be inlined.